cpp/src/arrow/dataset/test_util_internal.h

```diff
@@ -2140,8 +2140,8 @@ class WriteFileSystemDatasetMixin : public MakeFileSystemDatasetMixin {
         actual_struct = std::dynamic_pointer_cast<Array>(struct_array);
       }
 
-      auto expected_struct = ArrayFromJSON(struct_(expected_physical_schema_->fields()),
-                                           file_contents->second);
+      auto expected_struct = arrow::ArrayFromJSON(
```

Comment:
It's not obvious to me why you needed to add explicit `arrow::` prefixes here?

Reply:
You're right, nice catch. I can remove this. Is there an automated way to catch things like this?

Reply:
Not to my knowledge, no.

Reply:
Closed by PR: https://github.com/apache/arrow/pull/46695.